game/overlord/iso.cpp

```diff
@@ -658,11 +658,6 @@ u32 ISOThread() {
     ProcessMessageData();
 
     if (!read_buffer) {
-      // HACK!! sometimes when we want to exit, some other threads will wait for stuff to be loaded
-      // in such cases, we continue running until we're the last thread alive when it's safe to die
-      if (ThreadWantsExit(GetThreadId()) && OnlyThreadAlive(GetThreadId())) {
```

Comment:
Unfortunately I think this hack might be needed to shutdown if you close the window during a cutscene.

Reply:
What would go wrong in such a case? It seems to close properly.

Reply:
Try closing the window while a cutscene is playing while some music is fading out, that's what this fixed originally.

Reply:
I don't think it can be a problem now. If we're in this branch DelayThread will return control to the kernel and we'll end up at the exit condition check in iop_runner.